rfremoterunner/rf_server.py

```diff
@@ -0,0 +1,186 @@
+DEFAULT_PORT = 1471
+
+
+class RobotFrameworkServer:
+
+    # Executable RPC functions
+    EXECUTE_FUNC = 'execute_robot_run'
+
+    def __init__(self, address=DEFAULT_ADDRESS, port=DEFAULT_PORT, debug=False):
+        """
+        Constructor for RobotFrameworkServer
+
+        :param address: Address to bind to. Default is localhost
+        :type address: str
+        :param port: Port to listen on. Default is 1471
+        :type port: int
+        :param debug: Run in debug mode. This changes the logging level and does not cleanup the workspace
+        :type debug: bool
+        """
+        self._address = address
+        self._port = port
+        self._server = xmlrpc_server.SimpleXMLRPCServer((address, int(port)), encoding='utf-8')
+        self._server.register_function(RobotFrameworkServer.execute_robot_run, self.EXECUTE_FUNC)
+        logging_level = logging.DEBUG if debug else logging.INFO
+        logger.setLevel(logging_level)
+
+    def serve(self):
+        """
+        Blocking call to wait for XML-RPC connections
+        """
+        logger.info('Listening on {}:{}'.format(self._address, self._port))
+        self._server.serve_forever()
+
+    @staticmethod
+    def execute_robot_run(test_suites, dependencies, robot_args, debug=False):
+        """
+        Callback that is invoked when a request to execute a robot run is made
+
+        :param test_suites: Dictionary of suites to execute
+        :type test_suites: dict
+        :param dependencies: Dictionary of files that the test suites are dependant on
+        :type dependencies: dict
+        :param robot_args: Dictionary of arguments to pass to robot.run()
+        :type robot_args: dict
+        :param debug: Run in debug mode. This changes the logging level and does not cleanup the workspace
+        :type debug: bool
+
+        :return: Dictionary containing test results and artifacts
+        :rtype: dict
+        """
+        workspace_dir = None
+        std_out_err = None
+        old_cwd = None
+        try:
+            old_log_level = logger.level
+            if debug:
+                logger.setLevel(logging.DEBUG)
+
+            # Save all suites & dependencies to disk
+            workspace_dir = RobotFrameworkServer._create_workspace(test_suites, dependencies)
+
+            # Change the CWD to the workspace
+            old_cwd = os.getcwd()
+            os.chdir(workspace_dir)
+            sys.path.append(workspace_dir)
+
+            # Execute the robot run
+            std_out_err = StringIO()
+            logger.debug('Beginning Robot Run.')
+            logger.debug('Robot Run Args: ' + str(robot_args))
+            ret_code = run('.',
+                           stdout=std_out_err,
+                           stderr=std_out_err,
+                           outputdir=workspace_dir,
+                           name='Root',
+                           **robot_args)
+            logger.debug('Robot Run finished')
+            os.chdir(old_cwd)

```

Comment:
dont need this as doing in finally below

Reply:
Removed